### ISSUE TITLE
How to trigger a response given a focus change event 

### ISSUE THREAD

**Issue by DaveTJones** [crossterm-rs/crossterm#784]
Conditionally executing print statements as a result of FocusGained or FocusLost from within alacritty/gnome terminal doesn't produce any output when switching between panes. Based on the way things are presented within examples, this looks like it should be fairly straightforward.

To reproduce the issue, run the following script in a terminal (I used alacritty on Pop!_OS). Changing window focus doesn't result in printing:  
  
```
use crossterm::event::{read, Event};

fn print_events() {
    loop {
        match read() {
            Ok(Event::FocusGained) => println!("FocusGained"),
            Ok(Event::FocusLost) => println!("FocusLost"),
            _ => (),
        }
    }
}

fn main() {
    print_events();
}
````

How do I go about getting this to work? Or am I misunderstanding what these events represent?

**Comment by sigmaSd** [comment]
you need to enable focuschange https://github.com/crossterm-rs/crossterm/blob/master/examples/event-read.rs#L92

**Comment by DaveTJones** [comment]
Thanks a tonne, that fixed it. Although that example is pretty clear, the one I initially used from the docs was slightly misleading https://docs.rs/crossterm/latest/crossterm/event/index.html

**Comment by sigmaSd** [comment]
yeah that part could use a pr , there is a section about mouse events https://docs.rs/crossterm/latest/crossterm/event/index.html#mouse-events but its the same case with focus event and paste event

**Comment by DaveTJones** [comment]
I'll submit one later if that's cool?

**Comment by sigmaSd** [comment]
sure thanks

**Comment by DaveTJones** [comment]
Also noticed that this is missing from at least one other example. https://github.com/crossterm-rs/crossterm/blob/master/examples/event-poll-read.rs looks like it implements focus gain and loss events but is missing the relevant execute statements.
